app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -24,6 +25,12 @@ def create
 
     private
 
+    def check_personal_key_enabled
+      return if TwoFactorAuthentication::PersonalKeyPolicy.new(current_user).enabled?
```

Comment:
This will block the controller for users who don't have a personal key, but don't we also want to add a condition here for users who are IAL2?

Reply:
Nevermind, just saw your changes to the personal key policy. I missed those on my first pass :+1: